src/transformers/models/maskformer/configuration_maskformer.py

```diff
@@ -0,0 +1,179 @@
+
+    >>> # Initializing a MaskFormer facebook/maskformer-swin-base-ade configuration
+    >>> configuration = MaskFormerConfig()
+
+    >>> # Initializing a model from the facebook/maskformer-swin-base-ade style configuration
+    >>> model = MaskFormerModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```
+
+    """
+    model_type = "maskformer"
+    attribute_map = {"hidden_size": "mask_feature_size"}
+    backbones_supported = ["swin"]
+
+    def __init__(
+        self,
+        fpn_feature_size: int = 256,
+        mask_feature_size: int = 256,
+        no_object_weight: float = 0.1,
+        use_auxilary_loss: bool = False,
+        backbone_config: Optional[Dict] = None,
+        detr_config: Optional[Dict] = None,
+        init_std: float = 0.02,
+        init_xavier_std: float = 1.0,
+        dice_weight: float = 1.0,
+        cross_entropy_weight: float = 1.0,
+        mask_weight: float = 20.0,
+        num_labels: int = 150,
+        **kwargs,
+    ):
+        if backbone_config is None:
+            # fall back to https://huggingface.co/microsoft/swin-base-patch4-window12-384-in22k
+            backbone_config = SwinConfig(
+                image_size=384,
+                in_channels=3,
+                patch_size=4,
+                embed_dim=128,
+                depths=[2, 2, 18, 2],
+                num_heads=[4, 8, 16, 32],
+                window_size=12,
+                drop_path_rate=0.3,
+            )
+        else:
+            backbone_model_type = backbone_config.pop("model_type")
+            if backbone_model_type not in self.backbones_supported:
+                raise ValueError(
+                    f"Backbone {backbone_model_type} not supported, please use one of {','.join(self.backbones_supported)}"
+                )
+            backbone_config = AutoConfig.for_model(backbone_model_type, **backbone_config)
+
+        detr_config = DetrConfig() if detr_config is None else DetrConfig(**detr_config)
+
+        self.backbone_config = backbone_config
+        self.detr_config = detr_config
+        # main feature dimension for the model
+        self.fpn_feature_size = fpn_feature_size
+        self.mask_feature_size = mask_feature_size
+        # initializer
+        self.init_std = init_std
+        self.init_xavier_std = init_xavier_std
+        # Hungarian matcher && loss
+        self.cross_entropy_weight = cross_entropy_weight
+        self.dice_weight = dice_weight
+        self.mask_weight = mask_weight
+        self.use_auxilary_loss = use_auxilary_loss
+        self.no_object_weight = no_object_weight
+
+        self.num_attention_heads = self.detr_config.encoder_attention_heads
+        self.num_hidden_layers = self.detr_config.num_hidden_layers
+        super().__init__(num_labels=num_labels, **kwargs)
+
+    @classmethod
+    def from_backbone_and_detr_configs(cls, backbone_config: PretrainedConfig, detr_config: DetrConfig, **kwargs):
```

Comment:
Here we use the term "backbone" (not Swin) but the term "DETR".

Wondering if we can make it more general, like `transformer_decoder_config`.

Reply:
That's a great feedback, @sgugger commented that for now we can keep detr and change it later

Reply:
That was for non-user facing APIs. I think here we need to be more generic. We know we intend to support other decoder than DETR, so we should have a more agnostic name `from_backbone_and_decoder_configs` for instance, with `decoder_config` as argument.